src/services/positionService.ts

```diff
@@ -887,6 +889,95 @@ export default class PositionService {
     );
   }
 
+  async terminateManyRaw(positions: Position[]): Promise<TransactionResponse> {
+    const { chainId } = positions[0];
+
+    // Check that all positions are from the same chain
+    positions.forEach((position) => {
+      if (position.chainId !== chainId) {
+        throw new Error('Should not call terminate many for positions on different chains');
+      }
+    });
+
+    const companionInstance = await this.contractService.getHUBCompanionInstance(LATEST_VERSION);
+    const account = this.walletService.getAccount();
+    const terminatesPromises: Promise<PopulatedTransaction>[] = [];
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < positions.length; i++) {
+      const position = positions[i];
+      const hubAddress = await this.contractService.getHUBAddress(position.version);
+
+      const terminateData = companionInstance.populateTransaction.terminate(
```

Comment:
If you like, I can show you how to build transaction data without promises. Something like:
```
companionInstance.interface.encodeFunctionData('terminate', [
        hubAddress,
        position.positionId,
        account,
        account
    ]
)
```
This returns a `string`, instead of a `Promise<{data: string | undefined}>`